alphalens/plotting.py

```diff
@@ -13,20 +13,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import numpy as np
 import pandas as pd
 from scipy import stats
 import statsmodels.api as sm
 
-import seaborn as sns
-import matplotlib.cm as cm
-import matplotlib.pyplot as plt
-from matplotlib.ticker import ScalarFormatter
+import matplotlib
+if os.environ.get('DISPLAY', '') == '':
+    print('no display found. Using non-interactive Agg backend')
+    matplotlib.use('Agg')
+import seaborn as sns  # noqa: E402
+import matplotlib.cm as cm  # noqa: E402
+import matplotlib.pyplot as plt  # noqa: E402
+from matplotlib.ticker import ScalarFormatter  # noqa: E402
 
-from functools import wraps
+from functools import wraps  # noqa: E402
```

Comment:
Not a blocker, but maybe for later cleanup. I'd recommend adding a matplotlib rc that uses 'Agg', instead of doing it unconditionally at import time. Then we don't need all the noqa's either.

See https://github.com/quantopian/pyfolio/blob/master/pyfolio/tests/matplotlibrc and https://github.com/quantopian/pyfolio/blob/master/.travis.yml#L36